deeptime/markov/msm/tram/_tram.py

```diff
@@ -278,18 +240,28 @@ def fit(self, data, *args, **kw):
                                                                                                   bias_matrices)
         tram_input = tram.TRAMInput(state_counts, transition_counts, dtrajs, bias_matrices)
 
-        self._tram_estimator = tram.TRAM(tram_input)
-        self._run_estimation()
-        self._model = self._construct_markov_model()
+        if self._tram_estimator is None:
```

Comment:
What happens if I want to reuse the tram estimator with potentially different number of therm states and different number of markov states? I.e.: `tram.fit(some_data)` and later `tram.fit(other_data)`.

Reply:
Also I find it a bit odd that `n_therm_states` and `n_markov_states` are set instance-wide in a validation method

Reply:
Answer to comment 1: from the model, the number of therm/markov states are loaded. The data is validated to check whether all occuring state indices are smaller than the number of therm/markov states. So you can add data which is confined to less states than the original data, but not data that is expanded to more states.

Reply:
In that case I would structure things a bit differently. the number of therm/markov states is an attribute of the data but not so much of the estimator, so it should appear in the model but by dynamic otherwise. can you create the tram estimator instance inside fit?

Reply:
To comment 2: the number of states is obtained from the data (I could add the option that the numbers of states are supplied by the user). At the same time, the data is validated against the number of states (the bias matrix has to have the same dimension as the number of thermodynamic states, which is either `max(ttrajs)` or `len(dtrajs)` depending on what is given). So validation and determining the number of states are very much related. I split the method into three parts now so it's a bit clearer hopefully.